packages/generator-common/src/compiler.ts

```diff
@@ -0,0 +1,139 @@
+import { parse, resolve } from 'path';
+import { existsSync, promises } from 'fs';
+import { EOL } from 'os';
+import { createLogger } from '@sap-cloud-sdk/util';
+import {
+  CompilerOptions,
+  createProgram,
+  Diagnostic,
+  getPreEmitDiagnostics,
+  ModuleKind,
+  ModuleResolutionKind,
+  ScriptTarget
+} from 'typescript';
+import { GlobSync } from 'glob';
+
+const logger = createLogger('compiler');
+
+/**
+ * Executes the TypeScript compilation for the given directory.
+ * It recursively compiles all files ending with .ts
+ * @param path Directory to be compiled.
+ * @param compilerOptions Compiler options to be used
+ */
+export async function transpileDirectory(
+  path: string,
+  compilerOptions: CompilerOptions
+): Promise<void> {
+  logger.verbose(`Transpiling files in the directory: ${path} started.`);
+
+  const files = new GlobSync(resolve(path, '**/*.ts'), {}).found;
+  const program = await createProgram(files, compilerOptions);
+  const emitResult = await program.emit();
+
+  const allDiagnostics = getPreEmitDiagnostics(program).concat(
+    emitResult.diagnostics
+  );
+  if (allDiagnostics.length > 0) {
+    throw new Error(
+      `Compilation Errors:${EOL}${getErrorList(allDiagnostics).join(EOL)}`
+    );
+  }
+  logger.verbose(`Transpiling files in directory: ${path} finished.`);
+}
+
+function getErrorList(diagnostis: Diagnostic[]): string[] {
+  return diagnostis.map(
+    diagnostic =>
+      `${diagnostic.file?.fileName}:${diagnostic.start}:${diagnostic.length} - error TS${diagnostic.code}: ${diagnostic.messageText}`
```

Comment:
question:
depending on the value of `diagnostic.file?.fileName`, we use `diagnostic.start` or `diagnostic.length`.
However, there is only one test for the error message.
Could you please share more about `diagnostic.start` and `diagnostic.length`, because I see the type of these two are both number, while in the test, it's like a file path.
Maybe even add a test for the missing one?